### PR TITLE
Provide filled out item data when returning hold list

### PR DESCRIPTION
**Pull request by kenoir**
Put the full item into a holds list for the front-end.

Closes https://github.com/wellcomecollection/catalogue-api/issues/223

### CODE REVIEW

File: requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala [hunk 1]
```diff
@@ -21,16 +25,16 @@ trait LookupPendingRequests extends CustomDirectives {
...
 
         holdsWithCatalogueIds <- Future.sequence(
-          userHolds.right.get.holds.map { hold =>
+          userHolds.right.get.holds.map { hold: StacksHold =>
             itemLookup
               .bySourceIdentifier(hold.sourceIdentifier)
               .map {
-                case Left(elasticError) =>
-                  warn(s"Unable to look up $hold in Elasticsearch")
+                case Left(elasticError: ElasticsearchError) =>
```
Comment (kenoir) [inline requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala:32]: Previously we just chucked this error, not sure why - i've added it to the warning here.

File: common/stacks/src/main/scala/weco/api/stacks/models/display/DisplayResultsList.scala [hunk 2]
```diff
@@ -28,9 +28,11 @@ case class DisplayResultsList(
 object DisplayRequest {
   def apply(hold: StacksHold): DisplayRequest = {
     DisplayRequest(
-      item = new DisplayItem(
-        id = hold.canonicalId.map { _.underlying },
-        identifiers = Some(List(DisplayIdentifier(hold.sourceIdentifier)))
+      // TODO: This .get should always be Some here
```
Comment (kenoir) [inline common/stacks/src/main/scala/weco/api/stacks/models/display/DisplayResultsList.scala:31]: I'll refactor this next, but to speed up getting this data into the front-end we could merge this PR, deploy and put the refactor into a new PR.

File: requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala [hunk 3]
```diff
@@ -21,16 +25,18 @@ trait LookupPendingRequests extends CustomDirectives {
...
+                  warn(
+                    s"Unable to look up $hold in Elasticsearch: ${elasticError}"
```
Comment (alexwlchan) [inline requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala:34]: Let's not address it here for the sake of getting this PR out quickly, but having a sourceIdentifier on the hold feels a bit odd to me – a hold is a thing in Sierra and could have a source identifier, it's just not the identifier we're using here.

Comment (alexwlchan) [reply requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala:34]: What will happen in this branch if we can't find the item?

File: requests/src/test/scala/weco/api/requests/RequestsApiFeatureTest.scala [hunk 4]
```diff
@@ -98,6 +102,7 @@ class RequestsApiFeatureTest
              |            "type" : "Identifier"
              |          }
              |        ],
+             |        "title" : "${titleString}",
```
Comment (alexwlchan) [inline requests/src/test/scala/weco/api/requests/RequestsApiFeatureTest.scala:105]: nit:

```suggestion
             |        "title" : "$titleString",
```

**Review by alexwlchan — APPROVED** [review]

File: requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala (hunk 3 above)
Comment (kenoir) [reply requests/src/main/scala/weco/api/requests/responses/LookupPendingRequests.scala:34]: > but having a sourceIdentifier on the hold feels a bit odd to me

I agree, we use it to look up the item later and fill the Item option on `StacksHold` which itself feels wrong. I expect that it'll get removed refactoring out the `.get`.

> What will happen in this branch if we can't find the item?

Good point. I think this is actually a case that will blow up the `.get` on the option added in this PR (the`StacksHold` is returned with a `None` for an `Item`. It's still quite unlikely but we should handle this by doing a multi-item get and returning a list of the items we can get while logging that things are missing with a `warn`.

**Pull request merged**